File: tests/fixtures/python/classes.py
# Base class with __init__, methods, properties
class Animal:
    count = 0

    def __init__(self, name, sound):
        self.name = name
        self.sound = sound
        Animal.count += 1

    def speak(self):
        print(f"{self.name} says {self.sound}")
        return self.sound

    @classmethod
    def get_count(cls):
        return cls.count

    @staticmethod
    def is_animal(obj):
        return isinstance(obj, Animal)

    @property
    def display_name(self):
        return self.name.upper()


# Inherited class
class Dog(Animal):
    def __init__(self, name, breed):
        super().__init__(name, "Woof")
        self.breed = breed

    def speak(self):
        print(f"{self.name} the {self.breed} barks")
        return "Woof!"

    def fetch(self, item):
        return f"{self.name} fetches {item}"

File: tests/fixtures/typescript/classes.ts
// Base class with constructor, methods, and properties
class Animal {
  private name: string;
  protected sound: string;
  public static count: number = 0;

  constructor(name: string, sound: string) {
    this.name = name;
    this.sound = sound;
    Animal.count++;
  }

  getName(): string {
    return this.name;
  }

  speak(): string {
    console.log(`${this.name} says ${this.sound}`);
    return this.sound;
  }

  static getCount(): number {
    return Animal.count;
  }
}

// Extended class
class Dog extends Animal {
  private breed: string;

  constructor(name: string, breed: string) {
    super(name, "Woof");
    this.breed = breed;
  }

  speak(): string {
    console.log(`${this.getName()} the ${this.breed} barks`);
    return "Woof!";
  }

  fetch(item: string): string {
    return `${this.getName()} fetches ${item}`;
  }
}

export { Animal, Dog };
